### DISCUSSION TITLE
How to handle fixes for new features in pre-release branch?

### DISCUSSION THREAD

**Discussion by mklenbw** [semantic-release/semantic-release#3456] [Q&A]
Hey there,
i need help understanding how to use the prerelease workflow.

Let's say i have a prerelease branch containing a new feature. 

```
* feat: initial commit => v1.0.0 on @latest
| \
|   * feat: Cool new feature => v1.1.0-beta.1 on @beta
```

Now i recognized there's an issue with that feature and fixed it.

```
* feat: initial commit => v1.0.0 on @latest
| \
|   * feat: Cool new feature => v1.1.0-beta.1 on @beta
|   * fix: Fix for cool new feature => v1.1.0-beta.2 on @beta
```

If i merged that i would get both entries in my changelog, right?

```
* feat: initial commit => v1.0.0 on @latest
| \
|   * feat: Cool new feature => v1.1.0-beta.1 on @beta
|   * fix: Fix for cool new feature => v1.1.0-beta.2 on @beta
|  /
* merge: new feature => v1.1.0 on @latest
```

In reality i don't want the fix to show up because it's only valid for the prerelease. However i need that commit to get a prerelease out.

What's my best option here?

**Comment by travi** [comment]
you've made multiple valid changes that have impact to the consumers of your package. yes, semantic-release will identify each of them, leverage them to determine the appropriate version bump, and generate release notes. if you want to generate results outside of that process, your option is to edit your release notes manually after the release, which defeats some of the purpose of automating releases with semantic-release. part of developing software is that multiple changes are involved, our recommendation is simply to go ahead and communicate those to you consumers rather than trying to hide the process

**Reply by mklenbw** [reply]
That's the point. I did not do any changes for the customer.

Let's call it a release candidate that's not meant for public usage. 
How could this be a valid change to the feature if it isn't fully released yet?

I would end up having entries for little tweaks of the feature before even being released, right?
I guess from a customer perspective that's confusing.

**Reply by travi** [reply]
all releases are for a consumer. it might be limited release for a subset of early adopters to give feedback to you before promotion to a stable release, but once promoted, those are still valid changes that contributed to the resulting stable release. if you wish to communicate those changes differently, you need to manually modify the release notes.

if you are using the pre-release workflow as a way to test changes before a release, it is worth understanding that that is not an officially supported workflow for semantic-release. we provide a pre-release workflow, not a testing workflow to support approaches like gitflow. we have no intention to add behaviors to support using the pre-release workflow for enabling a testing workflow, which is not the intent of the pre-release workflow

**Comment by mklenbw** [comment]
What we do at the moment is to mark these commits with `[no-changelog]` and having a script filtering them out before releasing.
